core/domain/state_domain.py

```diff
@@ -1769,6 +1827,65 @@ def update_solicit_answer_details(self, solicit_answer_details):
                 % solicit_answer_details)
         self.solicit_answer_details = solicit_answer_details
 
+    def _get_all_translatable_content(self):
+        """Returns all content which can be translated into different languages.
+
+        Returns:
+            dict(str, str). Returns a dict with key as content id and content
+            html as the value.
+        """
+        content_id_to_html = {}
+
+        content_id_to_html[self.content.content_id] = self.content.html
+
+        # TODO(@DubeySandeep): Remove empty html checks once we add a validation
+        # check for each content in state should be non-empty html.
+        default_outcome = self.interaction.default_outcome
+        if default_outcome is not None and default_outcome.feedback.html != '':
+            content_id_to_html[default_outcome.feedback.content_id] = (
+                default_outcome.feedback.html)
+
+        for answer_group in self.interaction.answer_groups:
+            if answer_group.outcome.feedback.html != '':
+                content_id_to_html[answer_group.outcome.feedback.content_id] = (
+                    answer_group.outcome.feedback.html)
+
+        for hint in self.interaction.hints:
+            if hint.hint_content.html != '':
+                content_id_to_html[hint.hint_content.content_id] = (
+                    hint.hint_content.html)
+
+        solution = self.interaction.solution
+        if solution is not None and solution.explanation.html != '':
+            content_id_to_html[solution.explanation.content_id] = (
+                solution.explanation.html)
```

Comment:
Does all this logic duplicate any existing functionality? If so, try to reuse/consolidate code.